cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1091,20 +1091,19 @@ aggregate_reader_metadata::select_columns(
                                         has_list_parent || col_type == type_id::LIST);
         }
       } else {
-        for (size_t idx = 0; idx < col_name_info->children.size(); idx++) {
-          path_is_valid |=
-            build_column(&col_name_info->children[idx],
-                         find_schema_child(schema_elem, col_name_info->children[idx].name),
-                         output_col.children,
-                         has_list_parent || col_type == type_id::LIST);
+        for (const auto& idx : col_name_info->children) {
```

Comment:
Lots of west-const in this PR... can we autoformat to east const to match the rest of the library? (We don't have this option enabled in clang-format because it's supposedly bug-prone, but I don't think we observed any actual bugs when we last ran a manual formatting pass.)

Reply:
Hmm yeah at minimum I can do a manual pass on this PR and fix anything that goes wrong.

Reply:
Manual pass worked fine. [The warning is still present in the clang-format docs](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#qualifieralignment), though, so I've left the option out of our configuration.

Reply:
I dug up the original reviews for the PR where this feature and that warning were added. https://reviews.llvm.org/D69764#inline-1055542

Basically the breaking cases are things like:

```cpp
#define INTPTR int *

const INTPTR a;
```

where a macro would mess up the semantics of `const` and `*` by swapping the order.

I'd be okay with accepting that edge case and enabling this setting across the code base.

Reply:
OK, I'll add it for now. We can always disable it in the future if developers discover that it breaks things for them.